src/main/java/md/miller1995/Dealership/controllers/UserController.java

```diff
@@ -0,0 +1,25 @@
+package md.miller1995.Dealership.controllers;
+
+import md.miller1995.Dealership.securities.UserAuthDetails;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+
+
+@Controller
+@RequestMapping("/api/users")
+public class UserController {
+
+    @GetMapping()
+    public ResponseEntity<HttpStatus> getUserInfo(){
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        UserAuthDetails userAuthDetails = (UserAuthDetails) authentication.getPrincipal();
+        System.out.println(userAuthDetails.getUserEntity());
```

Comment:
I would move the logic in the `UserService` layer, inject that dependency in this controller class, and use a logger instead of a console print.
Try to make use of IoC and dependency injection to decouple modules, not the other way around.

Reply:
@AlexIliescu11 
This Controller and logic were built just for test, in the future I will remove or moved this method in another layer.

I added some new classes and some new implementations in the respective classes, can you please do a review ?